docs/src/main/asciidoc/rest-reactive.adoc

```diff
@@ -0,0 +1,1553 @@
+
+[source,java]
+----
+package org.acme.rest;
+
+import java.time.Duration;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+
+import io.smallrye.mutiny.Uni;
+
+@Path("yawn")
+public class Endpoint {
+
+    @GET
+    public Uni<String> blockingHello() throws InterruptedException {
+        return Uni.createFrom().item("Yaaaawwwwnnnnnn…")
+                // do a non-blocking sleep
+                .onItem().delayIt().by(Duration.ofSeconds(2));
+    }
+}
+----
+
+=== Exception mapping
+
+If your application needs to return non-nominal HTTP codes in error cases, the best is
+to throw exceptions that will result in the proper HTTP response being sent by the
+framework using link:{jaxrsapi}/javax/ws/rs/WebApplicationException.html[`WebApplicationException`] or any of its subtypes:
+
+[source,java]
+----
+package org.acme.rest;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.GET;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.Path;
+
+@Path("fromages/{fromage}")
+public class Endpoint {
+
+    @GET
+    public String findFromage(String fromage) {
+        if(fromage == null)
+            // send a 400
+            throw new BadRequestException();
+        if(!fromage.equals("camembert"))
+            // send a 404
+            throw new NotFoundException("Unknown cheese: " + fromage);
+        return "Camembert is a very nice cheese";
+    }
+}
+----
+
+If your endpoint method is delegating calls to another service layer which
+does not know of JAX-RS, you need a way to turn service exceptions to an
+HTTP response, and you can do that using the `@ServerExceptionMapper` annotation
+on a method, with one parameter of the exception type you want to handle, and turning
+that exception into a link:{jaxrsapi}/javax/ws/rs/core/Response.html[`Response`] (or a 
+link:{mutinyapi}/io/smallrye/mutiny/Uni.html[`Uni<Response>`]):
+
+[source,java]
+----
+package org.acme.rest;
+
+import java.util.Map;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+
+import org.jboss.resteasy.reactive.server.ServerExceptionMapper;
+
+class UnknownCheeseException extends RuntimeException {
+    public final String name;
+    
+    public UnknownCheeseException(String name) {
+        this.name = name;
+    }
+}
+
+@ApplicationScoped
+class CheeseService {
+    private static final Map<String, String> cheeses = 
+            Map.of("camembert", "Camembert is a very nice cheese",
+                   "gouda", "Gouda is acceptable too, especially with cumin");
+
+    public String findCheese(String name) {
+        String ret = cheeses.get(name);
+        if(ret != null)
+            return ret;
+        throw new UnknownCheeseException(name);
+    }
+}
+
+@Path("fromages/{fromage}")
+public class Endpoint {
+
+    @Inject
+    CheeseService cheeses;
```

Comment:
I could show that, but it's actually more verbose because I _still_ need a field, and not instead of an annotation I get an entire constructor. Do you think it's worth it?

Reply:
I do think it's worth it, because a lot of people still (10 years later) think that CDI and JAX-RS don't work properly with constructor injection :(

Reply:
OK

Reply:
Done.

Reply:
Are you sure? I still see the old code